Source/WebCore/platform/graphics/FontCascadeFonts.cpp

```diff
@@ -518,7 +523,7 @@ GlyphData FontCascadeFonts::glyphDataForCharacter(UChar32 c, const FontCascadeDe
     }).iterator->value;
 
     GlyphData glyphData = cacheEntry.glyphDataForCharacter(c);
-    if (!glyphData.glyph) {
+    if (!glyphData.font) {
```

Comment:
I think we'll always have a non-null font if we found a glyph (even if per your changes the glyph ID we chose is 0). And so I was going to suggest using `glyphData.isValid()`, for clarity, but reading that function makes me wonder if there are cases where the glyph ID is non-zero and the font is null, and that's a valid state for a `GlyphData` to have?

Reply:
That's a good question. I don't know if this would be a valid state. My interpretation was that if the font or the glyph are not zero, then the other will also be non-zero. However we can have the case where we explicitly set the glyph to be the .notdef glyph (glyph id = 0) but this is still a valid state.